### PR TITLE
feat(dlp): added a sample for deidentify table condition masking

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description

- Added a sample for [deidentify table condition masking](https://cloud.google.com/dlp/docs/samples/dlp-deidentify-table-condition-masking) with a unit test case.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/deid/deid_table_condition_masking.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		// Apply the condition to records
+		Expressions: &dlppb.RecordCondition_Expressions{
+			Type: &dlppb.RecordCondition_Expressions_Conditions{
+				Conditions: &dlppb.RecordCondition_Conditions{
+					Conditions: []*dlppb.RecordCondition_Condition{
+						{
+							Field:    fieldId,
+							Operator: dlppb.RelationalOperator_GREATER_THAN,
+							Value: &dlppb.Value{
+								Type: &dlppb.Value_IntegerValue{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_table_condition_masking.go:113]: nit: nesting is a bit deep here. You might create this above--especially if you're working with protobuf.Value types.

File: dlp/snippets/deid/deid_test.go [hunk 2]
```diff
@@ -157,6 +157,24 @@ func TestDeidentifyTableBucketing(t *testing.T) {
 
 }
 
+func TestDeidentifyTableMaskingCondition(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	buf := new(bytes.Buffer)
```
Comment (telpirion) [inline dlp/snippets/deid/deid_test.go:163]: issue: don't use `new` here. Instead, write `var buf bytes.Buffer`.

**Review by telpirion — CHANGES_REQUESTED** [review]

**Review by jakubrauch — APPROVED** [review]

File: dlp/snippets/deid/deid_table_condition_masking.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_table_condition_masking.go]: done.

File: dlp/snippets/deid/deid_test.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: Resolved.

**Review by telpirion — APPROVED** [review]

File: dlp/snippets/deid/deid_table_condition_masking.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	primitiveTransformation := &dlppb.PrimitiveTransformation_CharacterMaskConfig{
+		CharacterMaskConfig: charMaskConfig,
+	}
+
+	// Specify field to be de-identified.
+	fieldId := &dlppb.FieldId{
+		Name: "HAPPINESS SCORE",
+	}
+
+	recordCondition := &dlppb.RecordCondition_Condition{
```
Comment (jakubrauch) [inline dlp/snippets/deid/deid_table_condition_masking.go:102]: Could we have some comments on these, to clarif the meaning of these variables?

File: dlp/snippets/deid/deid_test.go [hunk 4]
```diff
@@ -174,35 +174,43 @@ func TestDeidentifyTableBucketing(t *testing.T) {
...
+		t.Fatal(err)
+	}
+	got := buf.String()
+	if want := "Table after de-identification :"; !strings.Contains(got, want) {
+		t.Errorf("deidentifyTableMaskingCondition got (%q) =%q ", got, want)
+	}
+	if want := "values:{string_value:\"**\"}"; !strings.Contains(got, want) {
+		t.Errorf("deidentifyTableMaskingCondition got (%q) =%q ", got, want)
+	}
+}
```
Comment (jakubrauch) [inline dlp/snippets/deid/deid_test.go:190]: Can we have an empty line after this?

File: dlp/snippets/deid/deid_test.go [hunk 5]
```diff
@@ -174,35 +174,43 @@ func TestDeidentifyTableBucketing(t *testing.T) {
...
 	tc := testutil.SystemTest(t)
-
 	var buf bytes.Buffer
 
 	if err := deidentifyTableConditionInfoTypes(&buf, tc.ProjectID, []string{"PATIENT", "FACTOID"}); err != nil {
 		t.Fatal(err)
 	}
 
 	got := buf.String()
-	if want := "Table after de-identification"; !strings.Contains(got, want) {
```
Comment (jakubrauch) [inline dlp/snippets/deid/deid_test.go:187]: Why do we delete these three lines?

File: dlp/snippets/deid/deid_table_condition_masking.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_table_condition_masking.go:103]: Done.

File: dlp/snippets/deid/deid_test.go (hunk 4 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go:190]: Done.

File: dlp/snippets/deid/deid_test.go (hunk 5 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: Resolved.

**Pull request merged**